src/mutation.cpp

```diff
@@ -835,28 +835,30 @@ void player::mutate_towards(std::string mut)
         mutation_effect(*this, mut);
         mutation_replaced = true;
     }
-    if (canceltrait != "") {
-        if(traits[mut].mixed_effect || traits[canceltrait].mixed_effect) {
-            rating = m_mixed;
-        } else if(traits[mut].points <= 0 && traits[canceltrait].points > 0) {
-            rating = m_good;
-        } else if(traits[mut].points > 0 && traits[canceltrait].points <= 0) {
-            rating = m_bad;
-        } else if(traits[mut].points == 0 && traits[canceltrait].points == 0) {
-            rating = m_neutral;
-        } else {
-            rating = m_mixed;
+    if (canceltrait.size() > 0) {
+        for (size_t i = 0; i < canceltrait.size(); i++) {
+            if(traits[mut].mixed_effect || traits[canceltrait[i]].mixed_effect) {
+                rating = m_mixed;
+            } else if(traits[mut].points <= 0 && traits[canceltrait[i]].points > 0) {
+                rating = m_good;
+            } else if(traits[mut].points > 0 && traits[canceltrait[i]].points <= 0) {
+                rating = m_bad;
+            } else if(traits[mut].points == 0 && traits[canceltrait[i]].points == 0) {
+                rating = m_neutral;
+            } else {
+                rating = m_mixed;
```

Comment:
So, if I'm reading this right, if the new mutation is a "good" mutation (say, `traits[mut].points == -3`), and the cancelled one is also a "good" mutation (say, `traits[canceltrait[i]].points == -1`), you get the mixed result. This seems reasonable at first glance, but is mutating "Night Vision" into "High Night Vision" really mixed?

Is there a reason just comparing the point values to each other doesn't work right? What I mean is, a comparison like `traits[mut].points > traits[canceltrait[i]].points`.